### ISSUE TITLE
Remote patches are not able to be installed on Windows

### ISSUE THREAD

**Issue by AlmightyTritan** [vaimo/composer-patches#33]
While trying to apply [the following patch](https://www.drupal.org/files/issues/2019-05-04/add-webform-composite-schema-to-make-fields-translatable-2979305-11.patch), the `composer patch:apply` command throws `Vaimo\ComposerPatches\Exceptions\PatchFailureException`. When I switch the WSL (Windows Subsystem for Linux) and run the command via bash the command passes perfectly.

**Pre-requisites**
1. Windows 10
2. WSL
3. vaimo/composer-patches with either version 4.11.1 or 4.16.1

**To Reproduce**
1. Add an externally hosted patch to the `composer.json`
   ```json
   "extra": {
       "patches": {
           "drupal/webform_composite": {
               "example": "https://www.drupal.org/files/issues/2019-05-04/add-webform-composite-schema-to-make-fields-translatable-2979305-11.patch"
           }
       }
   }
   ```
2. Run `composer install` or `composer patch:apply` in either Powershell or CMD
3. Run `composer install` or `composer patch:apply` in WSL bash

**Expected**
The patch will be applied to the `drupal/webform_composite` package

**Actual**
`Vaimo\ComposerPatches\Exceptions\PatchFailureException` is thrown with the error message of `Cannot apply patch C:\Users\TMCHAT~1\AppData\Local\Temp/composer-patches-9b6502bc15546a96fb461dbcc4363d53.patch`

**Notes**
I believe this may be caused by the paths not being normalized for the OS the command is run on. This is my current assumption as the error message seems to be using a linux directory separator between the Temp directory and the filename.

**Comment by allanpaiste** [comment]
Thank you so much for pointing this out. 

Issue caused indeed by https://github.com/vaimo/composer-patches/blob/f2e35be2f8db7b2a075747fb683f55609a7ffd5d/src/Patch/DefinitionList/LoaderComponents/DownloaderComponent.php#L60

**Comment by allanpaiste** [comment]
Addressed in [4.17.0 && 3.48.0](https://github.com/vaimo/composer-patches/blob/master/CHANGELOG.md#4170-2019-08-11)
